lib/selector.js

```diff
@@ -57,8 +59,35 @@ class Selector extends Prefixer {
     }
 
     let prefixeds = {}
-    for (let prefix of this.possible()) {
-      prefixeds[prefix] = this.replace(rule.selector, prefix)
+
+    if (rule.selector.includes(',')) {
```

Comment:
I completely lost the idea behind this method (but I wrote it :sob:).

Can you describe (you can do it in Russian) what do we do here?

Reply:
I'll try to do it in english :crossed_fingers: 

This method collects all possible prefixes for the selector, creates prefixed version of the ```rule.selector```, caches the result and adds it to the rule. So the rule looks like this at the end (copied from master branch):
```js
Rule {
    type: 'rule',
    selector: '.a, .a .b::placeholder, a:read-only',
    // lots of properties
    _autoprefixerPrefixeds: {
        '-webkit-': '.a, .a .b::-webkit-input-placeholder, a:read-only',
        '-moz-': '.a, .a .b::-moz-placeholder, a:read-only',
        '-ms-': '.a, .a .b::-ms-input-placeholder, a:read-only',
        '-o-': '.a, .a .b::-o-placeholder, a:read-only',
        '-moz- old': '.a, .a .b:-moz-placeholder, a:read-only',
        '-ms- old': '.a, .a .b:-ms-input-placeholder, a:read-only' }
    }
}
```
As you can see ```:read-only``` is left without prefixes.

Reply:
I refactored code, and I think now it's much more easier to reason about. Plus, I decided that each selector has to process only that part of rule that contains the selector and not the whole rule as I did before.